### PR TITLE
Add Dockerfile and exit 2 on verification failure

### PR DESCRIPTION
**Pull request by cnunciato**
This adds a Dockerfile and exits 2 when license_scout detects missing licenses, both of which we need in order to run this in our pipeline.

Signed-off-by: Christian Nunciato <cnunciato@chef.io>

### CODE REVIEW

File: bin/license_scout [hunk 1]
```diff
@@ -39,3 +39,5 @@ collector.run
 report = collector.issue_report
 
 puts report
+
+exit 2 if !report.empty?
```
Comment (danielsdeleo) [inline bin/license_scout]: Is there any verification we can do that this won't break any pipelines that are working or "working" now?

Comment (srenatus) [inline bin/license_scout]: I'd think that the change that drags this into automate's docker-compose.yml would be tested locally (as one of the goals of the DC based stuff was to be able to run everything locally...)

**Comment by jmink** [comment]
@danielsdeleo Have your concerns been addressed to the point where you're comfortable giving this a +1, and if not what needs to happen?

File: bin/license_scout (hunk 1 above)
Comment (cnunciato) [inline bin/license_scout]: @danielsdeleo Yeah, although I'm not sure what the best way would be to do that. One thing we could do is merge, then run an ad-hoc build and revert if it causes any trouble; if anyone knows of a way to run a build using this branch, though, that would be better.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
+FROM devchef/chefdk
+RUN apt-get update && apt-get install git -y
+
+COPY bin/ /usr/src/app/license_scout/bin/
+COPY lib/ /usr/src/app/license_scout/lib/
+COPY Gemfile Rakefile license_scout.gemspec /usr/src/app/license_scout/
+
+WORKDIR /usr/src/app/license_scout
+
+RUN bundle install --with=development
```
Comment (phiggins) [inline Dockerfile]: This should be `--without=development`. rspec and friends are not needed to run license_scout.

**Comment by phiggins** [comment]
It doesn't appear to be possible to use the dockerized version with a ruby project. The problem is license_scout expects to be able to shell out to bundler and get all of the gem metadata, which it can't do when it is isolated in its own docker container. https://github.com/chef/license_scout/blob/4f781af91c03e40db797c104a65612690a567d2d/lib/license_scout/dependency_manager/bundler.rb#L51

It might be possible to change how license_scout integrates with bundler to get this work: 
* Before running license_scount, run `bundle package` in the app to be checked.
* When license_scout is invoked, try to get the license data out of the gems now downloaded in the project's bundler cache directory.

I will attempt to that out to see if it is feasible.

**Comment by phiggins** [comment]
As an aside, it seems a bit weird for license_scout to have it's own Dockerfile so that it can integrate into another application's pipeline. Shouldn't that integration live with that project?

**Comment by blakestier** [comment]
I've unpinned this as it seems to still be WIP.

**Comment by cnunciato** [comment]
@phiggins Awesome, thanks -- feel free to push onto this branch as well. 

Including a Dockerfile is not uncommon, though; the presence of this file basically says to the user of this project, "If you want to run this as a container, here you go." In this case, one could pull down the repo and run:

```
docker build -t license_scout .
docker run -it --rm -v /path/to/myproject:/myproject license_scout /myproject
```

**Pull request merged**